scripts/tokenomics-emulation/nominators.ts

```diff
@@ -0,0 +1,189 @@
+import Accounts from "./accounts";
+import Network from "./network";
+import fs from "fs";
+
+class Nominator {
+  private stashAccount: any;
+  private controllerAccount: any;
+  private stashBalance: string;
+  private controllerBalance: string;
+
+  constructor(
+    private readonly network: Network,
+    private readonly accounts: Accounts,
+    private readonly decimals: number
+  ) {}
+
+  /**
+   * wait till node sync
+   * @param waitSeconds wait seconds
+   */
+  public async start(waitSeconds: string) {
+    console.log("Check if syncing...");
+    await this.callWithRetry(
+      this.isSyncing.bind(this),
+      {
+        maxDepth: 100,
+      },
+      0,
+      waitSeconds
+    );
+    console.log("Sync is complete!");
+  }
+
+  /**
+   * Load stash and controller accounts.
+   */
+  public async loadAccounts(stashMnemonic: string, controllerMnemonic: string) {
```

Comment:
Why this logic is in Nominator. It should be inside Accounts class.

Reply:
The core logic is in the accounts class only, here we are just assigning to the parameters.

Reply:
So, `addNominator` method requires the stash account. Just pass stash mnemonic as a parameter to it.

The same for `setController`. Pass controller mnemonic to it.

After you can remove the `loadAccounts`. The current API is not clear at all.

Reply:
Updated the code. removed load Accounts and passing stash and controller account as params